src/coll/mod.rs

```diff
@@ -46,9 +46,16 @@ const MAX_INSERT_DOCS_BYTES: usize = 16 * 1000 * 1000;
 ///
 /// ```rust
 /// # use bson::{bson, doc};
-/// # use mongodb::{Client, error::Result};
+/// # use mongodb::error::Result;
+/// # #[cfg(feature = "sync")]
```

Comment:
this doc example is in the async module, so I don't think we want to be using the sync client here. ditto for the others not in `src/sync`

Reply:
Good call; I've updated the docs for `Collection` and `Database`, but I left the event handling code docs as-is pending our decision on RUST-340.